paymentsheet/res/layout/fragment_primary_button_container.xml

```diff
@@ -1,10 +1,16 @@
 <?xml version="1.0" encoding="utf-8"?>
-<com.stripe.android.paymentsheet.ui.PrimaryButton xmlns:android="http://schemas.android.com/apk/res/android"
-    android:id="@+id/primary_button"
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
     android:layout_width="match_parent"
-    android:layout_height="@dimen/stripe_paymentsheet_primary_button_height"
-    android:layout_marginTop="@dimen/stripe_paymentsheet_button_container_spacing"
-    android:layout_marginStart="@dimen/stripe_paymentsheet_outer_spacing_horizontal"
-    android:layout_marginEnd="@dimen/stripe_paymentsheet_outer_spacing_horizontal"
-    android:text="@string/stripe_paymentsheet_pay_button_label"
-    android:visibility="gone" />
+    android:layout_height="wrap_content">
+    <!-- Needs to be wrapped in a FrameLayout to prevent layout issues after configuration changes. -->
```

Comment:
My guess is it's a compose bug. But I didn't look to far into it since the mitigation was low cost.

I've run into this before, which is why I suspected this and fixed it here. The layout inflator isn't taking into account the root view (from this XML file) layout params into account when inflating the view. An easy way around that is to not have any layout params in the root view, but keep them in the nested view, which is what I did here.